app/Http/Controllers/Api/UserController.php

```diff
@@ -25,9 +27,135 @@ public function __construct()
             $this->facultyId = $this->currentUser->faculty_id;
             $this->departmentId = $this->currentUser->department_id;
             $this->userId = $this->currentUser->id;
+            $this->roleId = Util::$roleId;
         }
     }
-    
+
+    public function createFacultyAdmin(Request $request)
+    {
+        $user = new User();
+
+        $validator = Validator::make($request->all(), [
+            'email' => 'required',
+            'password' => 'required',
+            'first_name' => 'required',
+            'faculty_id' => 'required',
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json(['error' => "All fields are required"], 400);
+        }
+
+        $id = Util::uuid();
+        $email = $request->get('email');
+        $password = md5($request->get('password'));
+        $first_name = $request->get('first_name');
+        $other_names = $request->get('other_names');
+        $faculty_id = $request->get('faculty_id');
+        $gender = $request->get('gender');
+        $user_ip_address = (new Util())->ip();
+        $status = $request->get('status') ?? 'Active';
+
+        $checkDuplicate = User::where('email', $email)->first();
+
+        if (empty($checkDuplicate)) {
+            $user->id = $id;
+            $user->username = $user->email = $email;
+            $user->password = $password;
+            $user->first_name = $first_name;
+            $user->other_names = $other_names;
+            $user->gender = $gender;
+            $user->user_ip_address = $user_ip_address;
+            $user->role_id = $this->roleId['faculty_admin'];
```

Comment:
Are we sure all users uploaded here are faculty admins @bynalab ? @saidabdul80 please confirm.

Reply:
This particular method is for faculty admin, students are uploaded into the system and school admins are manually created from our end. Was that what we agreed on?

Reply:
True..... So they question is, who is performing this action? School Admin?

Reply:
what i want is create user, please follow the interface pix i attached to see requirment,
this upload csv can have matric number as well.
because as you can see in the picture we have role, meaning student can also be upload @bynalab